### PR TITLE
feat: microphone setting

### PR DESCRIPTION
**Pull request by ryotarokato**

### CODE REVIEW

File: component/src/utils/HTTP/websocket.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export class Websocket {
   }
 
   public static createConnection(io: ServiceIO, messages: Messages) {
-    if (!document.body.contains(io.deepChat)) return; // check if element is still present
+    if (!(io.deepChat.getRootNode({composed: true}) instanceof Document)) return; // check if element is still present
     const websocketConfig = io.connectSettings.websocket;
     if (!websocketConfig) return;
     if (io.connectSettings.handler) return CustomHandler.websocket(io, messages);
```
Comment (EmberGlow01) [inline component/src/utils/HTTP/websocket.ts:27]: I think that we can use another method

File: component/src/utils/HTTP/websocket.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ export class Websocket {
 
   private static retryConnection(io: ServiceIO, messages: Messages) {
     io.deepChat._validationHandler?.();
-    if (!document.body.contains(io.deepChat)) return; // check if element is still present
+    if (!(io.deepChat.getRootNode({composed: true}) instanceof Document)) return; // check if element is still present
```
Comment (EmberGlow01) [inline component/src/utils/HTTP/websocket.ts:50]: This also

File: component/src/utils/HTTP/websocket.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ export class Websocket {
   }
 
   public static createConnection(io: ServiceIO, messages: Messages) {
-    if (!document.body.contains(io.deepChat)) return; // check if element is still present
+    if (!(io.deepChat.getRootNode({composed: true}) instanceof Document)) return; // check if element is still present
```
Comment (ryotarokato) [inline component/src/utils/HTTP/websocket.ts:24]: I think so, But now is OK

**Review by EmberGlow01 — APPROVED** [review]
LGTM

**Pull request merged**